cloudmarker/checks/firewallrulecheck.py

```diff
@@ -0,0 +1,119 @@
+"""Firewall Rule Check plugin for GCP Firewall rule record.
+
+This modules define a :class:`FirewallRuleCheck` that identifies weak Firewall
+rules for Google Cloud Platform.
+"""
+
+
+class FirewallRuleCheck:
+    """Firewall Rule Check plugin.
+
+    This class identifies weak Firewall rules in the record objects by
+    comparing the them with predefined weak Firewall rules
+    configurations.
+    """
+
+    def __init__(self):
+        """Create instance of :class:`FirewallRuleCheck` plugin."""
+        # self._misconfigurations defines the set of weak/misconfigured
+        # firewall rules. Given any firewall rule, is compared with
+        # these predefined weak misconfigurations if the rule matches to
+        # any one predefined rule, it will be marked as weak and will be
+        # reported as anomaly. Adding IPProtocol field have been skipped
+        # to check for misconfiguration as GCP docs says that: The
+        # protocol type is required when creating a firewall rule. This
+        # value can either be one of the following well known protocol
+        # strings (tcp, udp, icmp, esp, ah, ipip, sctp), or the IP
+        # protocol number, implying it cannot be *.
+        self._misconfigurations = {
+            'sourceRanges': [
+                '0.0.0.0/0',
+                '*'
+            ],
+            'destinationRanges': [
+                '0.0.0.0/0',
+                '*'
+            ],
+        }
+
+    def _process_rule(self, record):
+        _misconfigurations = []
+
+        for misconfig_field in self._misconfigurations:
+            if misconfig_field in record:
+                for record_misconfig_field in record[misconfig_field]:
+                    if (record_misconfig_field in
+                            self._misconfigurations[misconfig_field]):
+                        _misconfigurations.append(misconfig_field
+                                                  + ':'
+                                                  + record_misconfig_field)
+        return {
+            'misconfigurations': _misconfigurations,
+            'id': record['id']
+        }
+
+    def eval(self, record):
+        """Evaluate Firewall rules for a given record.
+
+        Evaluate the weak security misconfigurations for each record and
+        collate each weakly identified record with its misconfigurations. This
+        function will only evaluate firewall_rule record.
+
+        For each such record it will create a dict object containing the weak
+        misconfigurations and return those records.
+
+        Arguments:
+            record (dict): Dictionary containing the rules definition.
+
+        Yields:
+            anomaly (dict): Dictionary representing anomaly.
+
+        The anomaly record yielded by this method would look like:
+
+        .. code:: python
+
+            {
+                "record_type": "firewall_rule_anomaly",
+                "id": ""
```

Comment:
Please remove this line for `"id"` because I see there is a line for `"id"` a few lines below (line 86).